utils/swift_build_support/swift_build_support/xcrun.py

```diff
@@ -10,25 +10,43 @@
 #
 # ----------------------------------------------------------------------------
 #
-# Python wrappers for invoking `xcrun` on the command-line.
+# Python wrappers for invoking `xcrun` and `xcodebuild` on the command-line.
 #
 # ----------------------------------------------------------------------------
 
 import subprocess
 
 
+def _output(args):
+    try:
+        out = subprocess.check_output(args, stderr=subprocess.PIPE)
+        return out.rstrip()
+    except subprocess.CalledProcessError:
+        return None
+
+
 def find(toolchain, tool):
     """
     Return the path for the given tool, according to `xcrun --find`, using
     the given toolchain. If `xcrun --find` cannot find the tool, return None.
     """
-    try:
-        # `xcrun --find` prints to stderr when it fails to find the
-        # given tool. We swallow that output with a pipe.
-        out = subprocess.check_output(['xcrun', '--sdk', 'macosx',
-                                       '--toolchain', toolchain,
-                                       '--find', tool],
-                                      stderr=subprocess.PIPE)
-        return out.rstrip()
-    except subprocess.CalledProcessError:
-        return None
+    return _output(['xcrun', '--sdk', 'macosx',
+                    '--toolchain', toolchain, '--find', tool])
+
+
+def xcodebuild_version():
+    """
+    Return a string with version information for the host machine's
+    `xcodebuild`. If `xcodebuild` is not installed, return None.
+    """
+    return _output(['xcodebuild', '-version'])
+
+
+def xcodebuild_sdk_version(sdk):
```

Comment:
I somehow feel like abstracting these invocations of `xcodebuild` does not abstract anything.  Maybe inline them into their only caller for now?